pkg/infra/iac2/templates_compiler.go

```diff
@@ -188,78 +197,133 @@ func (tc TemplatesCompiler) resolveDependencies(resource core.Resource) string {
 }
 
 // resolveStructInput translates a value to a form suitable to inject into the typescript as an input to a function.
-func (tc TemplatesCompiler) resolveStructInput(childVal reflect.Value, useDoubleQuotedStrings bool) string {
+func (tc TemplatesCompiler) resolveStructInput(childVal reflect.Value, useDoubleQuotedStrings bool, iacTag string) (string, error) {
 	var zeroValue reflect.Value
 	if childVal == zeroValue {
-		return `null`
+		return `null`, nil
 	}
 	switch childVal.Kind() {
 	case reflect.Bool,
 		reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64,
 		reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64,
 		reflect.Float32, reflect.Float64:
-		return fmt.Sprintf("%v", childVal.Interface())
+		return fmt.Sprintf("%v", childVal.Interface()), nil
 	case reflect.String:
-		return quoteTsString(childVal.Interface().(string), useDoubleQuotedStrings)
+		return quoteTsString(childVal.Interface().(string), useDoubleQuotedStrings), nil
 	case reflect.Struct, reflect.Pointer:
 		if childVal.Kind() == reflect.Pointer && childVal.IsNil() {
-			return "null"
+			return "null", nil
 		}
 		if typedChild, ok := childVal.Interface().(core.Resource); ok {
-			return tc.getVarName(typedChild)
+			return tc.getVarName(typedChild), nil
 		} else if typedChild, ok := childVal.Interface().(core.IaCValue); ok {
-			return tc.handleIaCValue(typedChild)
+			output, err := tc.handleIaCValue(typedChild)
+			if err != nil {
+				return output, err
+			}
+			return output, nil
 		} else {
-			return ""
+			if iacTag == "document" {
+				output := "{"
+				correspondingStruct := childVal
+				if childVal.Kind() == reflect.Pointer {
+					correspondingStruct = childVal.Elem()
+				}
+				for i := 0; i < correspondingStruct.NumField(); i++ {
+
+					anotherChildVal := correspondingStruct.Field(i)
+					fieldName := correspondingStruct.Type().Field(i).Name
+					resolvedValue, err := tc.resolveStructInput(anotherChildVal, false, iacTag)
+
+					if err != nil {
+						return output, err
+					}
+					if resolvedValue == "" {
+						return output, errors.Errorf(`child struct of %v is not of a known type: %v`, childVal.Type().Name(), fieldName)
```

Comment:
so the way to change this if you dont want to handle both is returning the error on structs we dont know how to parse. Otherwise we would never catch it for arrays and maps since they dont return `""` to our top level function and thats the only thing that signals an error today. I can go ahead and do that though